core/src/main/java/hudson/model/User.java

```diff
@@ -813,6 +837,21 @@ static File getRootDir() {
         return new File(Jenkins.get().getRootDir(), "users");
     }
 
+    private static final int PREFIX_MAX = 14;
```

Comment:
Original version in `UserIdMapper` set this to 15 but then subtracted one for the `_`. I find this a bit clearer. Of course the size is arbitrary; here it allows directory names up to 79 (ASCII) characters, which I think is safe on all operating systems. The bigger concern on Windows is a _total_ path length approaching (IIRC) 254, when not using the UNC `\\?\` trick, but if you control the `%JENKINS_HOME%` length, this should not be too bad: `C:\Jenkins\users\quixotichacker_01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b\config.xml` is only 107.

Reply:
> The bigger concern on Windows is a total path length approaching (IIRC) 254, when not using the UNC \\?\ trick, but if you control the %JENKINS_HOME% length, this should not be too bad:

Java has been using the unicode API with `\\?\` for a loooong time. 
https://github.com/openjdk/jdk21u-dev/blame/d90297a828dff468afc34e2767439e51379f4f95/src/java.base/windows/classes/sun/nio/fs/WindowsPath.java#L174-L178

Reply:
OK, I just recall problems with `branch-api` on Windows.